packages/discord-player/src/DefaultVoiceStateHandler.ts

```diff
@@ -62,6 +62,7 @@ export async function defaultVoiceStateHandler(player: Player, queue: GuildQueue
                     if (!player.nodes.has(queue.guild.id)) return;
                     if (queue.options.leaveOnEmpty) queue.delete();
                     player.events.emit(GuildQueueEvent.emptyChannel, queue);
+                    player.events.emit(GuildQueueEvent.playerFinish, queue);
```

Comment:
`playerFinish` indicates the AudioPlayer's lifecycle thus it should not be emitted from here. The reason why this event is not emitted is because `queue.delete()` prevents events from being dispatched, unless it is emitted from `player.events` object. Most of the internals of `GuildQueue` make the use of `queue.emit()` method, which skips the event if the queue is deleted.

Reply:
if its not from here , shouldn't it be emitted somewhere?

Reply:
It is being emitted from here:

https://github.com/Androz2091/discord-player/blob/3abe3858d1075658369518b09528b4488e3c54ee/packages/discord-player/src/manager/GuildQueue.ts#L919

but like I said, it uses `this.emit()` instead of `this.player.events.emit()` which skips the event dispatch if the queue is deleted. I guess it makes sense to use `this.player.events.emit()` here since it indicates `AudioPlayer` lifecycle, which is pretty reasonable even for deleted queues (in other words, queue deletion occurs before this event is dispatched).

Reply:
It is also possible for the queue to enter `TransitionMode` upon calling `queue.delete()`, which also prevents audio player events from being dispatched.

Reply:
`AudioPlayer` is guaranteed to stop upon calling `queue.delete()` since it destroys the voice connection as well, making the entire queue object unusable. Therefore, it is safe for you to assume the `AudioPlayer` has stopped at this stage. **In other words, it is not necessary to emit `playerFinish` here.**

Reply:
hmm that seems fair. sorry for the trouble!